features/step_definitions/node.rb

```diff
@@ -52,6 +52,27 @@
   cache_resources *cb[cbname].shuffle
 end
 
+Given /^I select a random( schedulable| ready and schedulable)? (node|master|worker) not named #{QUOTED}$/ do |state, role, name|
+  ensure_admin_tagged
+
+  nodes = env.nodes.select { |n| n.name != name }
+
+  if state != nil && state.strip == "schedulable"
+    nodes = nodes.select { |n| n.schedulable? }
+  else
+    nodes = nodes.select { |n| n.ready? && n.schedulable? }
+  end
+
+  if role == "worker"
+    nodes = nodes.select { |n| n.is_worker? }
+  elsif role == "master"
+    nodes = nodes.select { |n| n.is_master? }
+  end
+
+  cache_resources *nodes
+  @host = nodes.sample.host
+end
```

Comment:
we have a step `I store the( schedulable| ready and schedulable)? (node|master|worker)s in the#{OPT_SYM} clipboard`. You can use it to store nodes. Then you use step `I use the #{QUOTED} node` to use one node and then another.

Having as separate step, I'm not sure is needed. Do you see any advantage wrt funcitonality or readability?

Reply:
You can see examples where `I store the( schedulable| ready and schedulable)? (node|master|worker)s in the#{OPT_SYM} clipboard` is already used.

Reply:
It would be nice to combine, but I need the exclusion,  `n.name != name` so I can select a node other than the OVN leader.  

Another way could be a `.filter{ |n| n.name != name}` or something applied to the clipboard nodes.
Or maybe a tweak of the select a random node step.

Reply:
Got you. How about adding optional `( excluding #{QUOTED})?` to the existing select step?
I'm a little reluctant to add too much complication to steps but adding a separate almost exact copy step to add same complication is perhaps is no better I guess.

Reply:
modified step to be  `Given /^I store the( schedulable| ready and schedulable)? (node|master|worker)s in the#{OPT_SYM} clipboard(?: excluding #{QUOTED})?$/`

it seems to work, needs more testing.

Reply:
Cool, if step works, pls remove this one.

Reply:
removed old step